google/cloud/spanner_v1/database.py

```diff
@@ -1038,7 +1068,9 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                     "CommitStats: {}".format(self._batch.commit_stats),
                     extra={"commit_stats": self._batch.commit_stats},
                 )
-            self._database._pool.put(self._session)
+            if self._batch._session is not None:
+                self._database._pool.put(self._session)
+            self._session._transaction = None
```

Comment:
We are also deleting this in commit and rollback methods, apart from all these methods.  https://github.com/googleapis/python-spanner/blob/main/google/cloud/spanner_v1/transaction.py#L181
https://github.com/googleapis/python-spanner/blob/main/google/cloud/spanner_v1/transaction.py#L246

Also In case of retry due to abort errors. https://github.com/googleapis/python-spanner/blob/main/google/cloud/spanner_v1/session.py#L388